xskillscore/core/contingency.py

```diff
@@ -796,3 +796,222 @@ def _gerrity_s(table):
+            dichotomous_category_edges,
+            dim=dim,
+        )
+        fpr.append(dichotomous_contingency.false_alarm_rate())
+        tpr.append(dichotomous_contingency.hit_rate())
+    tpr = xr.concat(tpr, "probability_bin")
+    fpr = xr.concat(fpr, "probability_bin")
+
+    easyfillna = True
+    if easyfillna:
+        fpr = fpr.fillna(1.0)
+        tpr = tpr.fillna(0.0)
+
+    # pad (0,0) and (1,1)
+    fpr_pad = xr.concat(
+        [
+            xr.ones_like(fpr.isel(probability_bin=0, drop=True)),
+            fpr,
+            xr.zeros_like(fpr.isel(probability_bin=-1, drop=True)),
+        ],
+        "probability_bin",
+    )
+    tpr_pad = xr.concat(
+        [
+            xr.ones_like(tpr.isel(probability_bin=0, drop=True)),
+            tpr,
+            xr.zeros_like(tpr.isel(probability_bin=-1, drop=True)),
+        ],
+        "probability_bin",
+    )
+
+    # https://github.com/scikit-learn/scikit-learn/blob/42aff4e2edd8e8887478f6ff1628f27de97be6a3/sklearn/metrics/_ranking.py#L916
+    # Attempt to drop thresholds corresponding to points in between and
+    # collinear with other points. These are always suboptimal and do not
+    # appear on a plotted ROC curve (and thus do not affect the AUC).
+    # Here np.diff(_, 2) is used as a "second derivative" to tell if there
+    # is a corner at the point. (...) This keeps all cases where the point should be
+    # kept, but does not drop more complicated cases like fps = [1, 3, 7],
+    # tps = [1, 2, 4]; there is no harm in keeping too many thresholds.
+    if drop_intermediate and fpr.probability_bin.size > 2:
+
+        def _drop_intermediate(fpr, tpr):
+            if isinstance(fpr, xr.Dataset):
+                if len(fpr.data_vars) == 1:
+                    v = list(fpr.data_vars)[0]
+                    tpr_check = tpr[v]
+                    fpr_check = fpr[v]
+                else:
+                    raise ValueError(
+                        "drop_intermediate=True only works for one variable xr.Dataset or xr.DataArray."
+                    )
+            else:
+                tpr_check = tpr
+                fpr_check = fpr
+            optimal_idxs = np.where(
+                np.r_[
+                    True,
+                    np.logical_or(
+                        fpr_check.diff("probability_bin", 2),
+                        tpr_check.diff("probability_bin", 2),
+                    ),
+                    True,
+                ]
+            )[0]
+            tpr = tpr.isel(probability_bin=optimal_idxs)
+            fpr = fpr.isel(probability_bin=optimal_idxs)
+            return fpr, tpr
+
+        fpr, tpr = _drop_intermediate(fpr, tpr)
+        fpr_pad, tpr_pad = _drop_intermediate(fpr_pad, tpr_pad)
+
+    def auc(tpr, fpr, dim="probability_bin"):
+        """Get area under the curve with trapez method."""
+        area = xr.apply_ufunc(
+            np.trapz, -tpr, fpr, input_core_dims=[[dim], [dim]], dask="allowed"
```

Comment:
Why is this `-tpr` and not `tpr`?

Reply:
Also, does this work correctly for multidimensional data, i.e. when `dim` is a list?

Reply:
it does, but not `drop_intermediate`

Reply:
If that's the case, could you add a comment to the docstring about `drop_intermediate` not working for multiple dims?